tasks/activate_account_user.py

```diff
@@ -0,0 +1,55 @@
+import logging
+
+from shared.celery_config import activate_account_user_task_name
+from shared.django_apps.codecov_auth.models import Account, Owner
+
+from app import celery_app
+from tasks.base import BaseCodecovTask
+
+log = logging.getLogger(__name__)
+
+
+class ActivateAccountUserTask(BaseCodecovTask, name=activate_account_user_task_name):
+    def run_impl(
+        self,
+        *,
+        user_ownerid: int,
+        org_ownerid: int,
+        **kwargs,
+    ):
+        """
+        Runs the task to activate a user onto an account.
+        :param user_ownerid: the user's owner id
+        :param org_ownerid: the organization owner id
+        """
+        log_context = {"user_ownerid": user_ownerid, "org_ownerid": org_ownerid}
+        log.info(
+            "Syncing account for user",
+            extra=log_context,
+        )
+
+        owner_user: Owner = Owner.objects.get(pk=user_ownerid)
+
+        # NOTE: We're currently ignoring organizations that don't have an account.
+        org_owner: Owner = Owner.objects.get(pk=org_ownerid)
+        account: Account | None = org_owner.account
+        if not account:
+            log.info(
+                "Organization does not have an account. Skipping account user activation."
+            )
+            return {"successful": True}
+
+        if account.can_activate_user(owner_user.user):
+            account.activate_owner_user_onto_account(owner_user)
+            account.save()
```

Comment:
I think we should add a log here to catch the condition when the org _does_ have an account but can_activate_user comes back **False** - what do you think?
and if this situation occurs - is the user_owner left in an in-between state or one that we would need to correct later?

Reply:
> is the user_owner left in an in-between state or one that we would need to correct later?

Yes, potentially. Unfortunately, the auto-activation code are raw SQL or a Postgres function. If we did our checks [in the activation code](https://github.com/codecov/worker/blob/main/services/activation.py), then it would be easier to keep them in sync. Anyways, I'll log when this is `False` it for now and we'll see how often this gets out of sync. To fix this properly, I'll probably want to do a SQLAlchemy/raw SQL -> Django migration for the activation stuff (as well as the new user activation task possibly). Maybe this can be a followup of this change...

Reply:
Added an info log. I think that the most likely scenario for this to be False is because the user has already been added. So, going to just keep the message level as INFO.